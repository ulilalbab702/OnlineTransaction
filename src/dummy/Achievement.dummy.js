export const Achievement = [
    {
        "Code": "FM-JBYPM",
        "Name": "FMC Jembayan PAMA",
        "Percent": 50,
        "BackLogOpen": 1,
        "BackLogClose": 1,
        "Priority1": 1,
        "Priority2": 0,
        "Priority3": 1,
        "Priority4": 0
    }
]