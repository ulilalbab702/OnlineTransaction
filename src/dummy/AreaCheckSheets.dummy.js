export const AreaChecksheets = [
    {
        "ZoneDesc": "ZONE 1 FRONT & UPPER AREA",
        "Area": "A. FRONT SIDE OF MACHINE CHECK",
        "GoodCondition": 9,
        "BadCondition": 0,
        "UncheckCondition": 0,
        "CheckSheetValues": [
            {
                "WoId": "47284214-faf8-464f-a967-d26a8cf0675e",
                "BacklogId": null,
                "WoNumber": "70156080",
                "UnitModel": "HD785-7",
                "SerialNumber": "9402",
                "UnitCode": "DT4112",
                "EquipmentNumber": "106921",
                "FormNumber": "HD785-7/3402/00/QA10/13(5)",
                "ZoneDesc": "ZONE 1 FRONT & UPPER AREA",
                "AreaDesc": "A. FRONT SIDE OF MACHINE CHECK",
                "ItemDesc": "Abnormal noise and discuss with operator (get opinion)",
                "Version": "1.00",
                "Measurement": "0",
                "Sequence": 1,
                "MPriority": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MNote": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MCondition": {
                    "Measurement": null,
                    "TypeDesc": "0",
                    "Code": "NMG",
                    "Desc": "Good",
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "Comment": null,
                "LeaderApprovedBy": "80101070@unitedtractors.com",
                "LeaderApprovedDate": "2020-04-21T10:02:03.291",
                "SupervisorApprovedDate": null,
                "SupervisorApprovedBy": null,
                "Id": "9279e283-45f9-5df0-805f-64a5469eda3a",
                "CreatedBy": "81300006@unitedtractors.com",
                "CreatedDate": "2020-04-21T09:49:49.894",
                "UpdatedBy": "81300006@unitedtractors.com",
                "UpdatedDate": "2020-04-21T09:49:49.894",
                "IsActive": true
            },
            {
                "WoId": "47284214-faf8-464f-a967-d26a8cf0675e",
                "BacklogId": null,
                "WoNumber": "70156080",
                "UnitModel": "HD785-7",
                "SerialNumber": "9402",
                "UnitCode": "DT4112",
                "EquipmentNumber": "106921",
                "FormNumber": "HD785-7/3402/00/QA10/13(5)",
                "ZoneDesc": "ZONE 1 FRONT & UPPER AREA",
                "AreaDesc": "A. FRONT SIDE OF MACHINE CHECK",
                "ItemDesc": "Shutdown switch & lockout switch function and condition.",
                "Version": "1.00",
                "Measurement": "0",
                "Sequence": 2,
                "MPriority": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MNote": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MCondition": {
                    "Measurement": null,
                    "TypeDesc": "0",
                    "Code": "NMG",
                    "Desc": "Good",
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "Comment": null,
                "LeaderApprovedBy": "80101070@unitedtractors.com",
                "LeaderApprovedDate": "2020-04-21T10:02:03.292",
                "SupervisorApprovedDate": null,
                "SupervisorApprovedBy": null,
                "Id": "613925d4-bb71-5baa-8eae-be0f56fa3acf",
                "CreatedBy": "81300006@unitedtractors.com",
                "CreatedDate": "2020-04-21T09:49:49.904",
                "UpdatedBy": "81300006@unitedtractors.com",
                "UpdatedDate": "2020-04-21T09:49:49.904",
                "IsActive": true
            },
            {
                "WoId": "47284214-faf8-464f-a967-d26a8cf0675e",
                "BacklogId": null,
                "WoNumber": "70156080",
                "UnitModel": "HD785-7",
                "SerialNumber": "9402",
                "UnitCode": "DT4112",
                "EquipmentNumber": "106921",
                "FormNumber": "HD785-7/3402/00/QA10/13(5)",
                "ZoneDesc": "ZONE 1 FRONT & UPPER AREA",
                "AreaDesc": "A. FRONT SIDE OF MACHINE CHECK",
                "ItemDesc": "Radiator fin and grille, unit code no. , after cooler, ac condenser,  oil cooler, hoses and mounting bolts check condition.",
                "Version": "1.00",
                "Measurement": "0",
                "Sequence": 3,
                "MPriority": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MNote": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MCondition": {
                    "Measurement": null,
                    "TypeDesc": "0",
                    "Code": "NMG",
                    "Desc": "Good",
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "Comment": null,
                "LeaderApprovedBy": "80101070@unitedtractors.com",
                "LeaderApprovedDate": "2020-04-21T10:02:03.292",
                "SupervisorApprovedDate": null,
                "SupervisorApprovedBy": null,
                "Id": "cd499279-41de-5b07-acdb-1111ab54472e",
                "CreatedBy": "81300006@unitedtractors.com",
                "CreatedDate": "2020-04-21T09:49:49.912",
                "UpdatedBy": "81300006@unitedtractors.com",
                "UpdatedDate": "2020-04-21T09:49:49.912",
                "IsActive": true
            },
            {
                "WoId": "47284214-faf8-464f-a967-d26a8cf0675e",
                "BacklogId": null,
                "WoNumber": "70156080",
                "UnitModel": "HD785-7",
                "SerialNumber": "9402",
                "UnitCode": "DT4112",
                "EquipmentNumber": "106921",
                "FormNumber": "HD785-7/3402/00/QA10/13(5)",
                "ZoneDesc": "ZONE 1 FRONT & UPPER AREA",
                "AreaDesc": "A. FRONT SIDE OF MACHINE CHECK",
                "ItemDesc": "Work lamp function and condition",
                "Version": "1.00",
                "Measurement": "0",
                "Sequence": 4,
                "MPriority": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MNote": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MCondition": {
                    "Measurement": null,
                    "TypeDesc": "0",
                    "Code": "NMG",
                    "Desc": "Good",
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "Comment": null,
                "LeaderApprovedBy": "80101070@unitedtractors.com",
                "LeaderApprovedDate": "2020-04-21T10:02:03.292",
                "SupervisorApprovedDate": null,
                "SupervisorApprovedBy": null,
                "Id": "d945f778-f49e-52d3-8b1f-a3365a3ba4eb",
                "CreatedBy": "81300006@unitedtractors.com",
                "CreatedDate": "2020-04-21T09:49:49.921",
                "UpdatedBy": "81300006@unitedtractors.com",
                "UpdatedDate": "2020-04-21T09:49:49.921",
                "IsActive": true
            },
            {
                "WoId": "47284214-faf8-464f-a967-d26a8cf0675e",
                "BacklogId": null,
                "WoNumber": "70156080",
                "UnitModel": "HD785-7",
                "SerialNumber": "9402",
                "UnitCode": "DT4112",
                "EquipmentNumber": "106921",
                "FormNumber": "HD785-7/3402/00/QA10/13(5)",
                "ZoneDesc": "ZONE 1 FRONT & UPPER AREA",
                "AreaDesc": "A. FRONT SIDE OF MACHINE CHECK",
                "ItemDesc": "Rotary (Accessories) check function",
                "Version": "1.00",
                "Measurement": "0",
                "Sequence": 5,
                "MPriority": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MNote": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MCondition": {
                    "Measurement": null,
                    "TypeDesc": "0",
                    "Code": "NMG",
                    "Desc": "Good",
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "Comment": null,
                "LeaderApprovedBy": "80101070@unitedtractors.com",
                "LeaderApprovedDate": "2020-04-21T10:02:03.292",
                "SupervisorApprovedDate": null,
                "SupervisorApprovedBy": null,
                "Id": "e5789900-adfb-52b5-9e67-8fc57faf93ab",
                "CreatedBy": "81300006@unitedtractors.com",
                "CreatedDate": "2020-04-21T09:49:49.932",
                "UpdatedBy": "81300006@unitedtractors.com",
                "UpdatedDate": "2020-04-21T09:49:49.932",
                "IsActive": true
            },
            {
                "WoId": "47284214-faf8-464f-a967-d26a8cf0675e",
                "BacklogId": null,
                "WoNumber": "70156080",
                "UnitModel": "HD785-7",
                "SerialNumber": "9402",
                "UnitCode": "DT4112",
                "EquipmentNumber": "106921",
                "FormNumber": "HD785-7/3402/00/QA10/13(5)",
                "ZoneDesc": "ZONE 1 FRONT & UPPER AREA",
                "AreaDesc": "A. FRONT SIDE OF MACHINE CHECK",
                "ItemDesc": "LH & RH Front Turn Lamp/Clearance Lamp check function.",
                "Version": "1.00",
                "Measurement": "0",
                "Sequence": 6,
                "MPriority": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MNote": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MCondition": {
                    "Measurement": null,
                    "TypeDesc": "0",
                    "Code": "NMG",
                    "Desc": "Good",
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "Comment": null,
                "LeaderApprovedBy": "80101070@unitedtractors.com",
                "LeaderApprovedDate": "2020-04-21T10:02:03.293",
                "SupervisorApprovedDate": null,
                "SupervisorApprovedBy": null,
                "Id": "975408ee-54aa-5764-bb5a-137b7002e9da",
                "CreatedBy": "81300006@unitedtractors.com",
                "CreatedDate": "2020-04-21T09:49:49.94",
                "UpdatedBy": "81300006@unitedtractors.com",
                "UpdatedDate": "2020-04-21T09:49:49.94",
                "IsActive": true
            },
            {
                "WoId": "47284214-faf8-464f-a967-d26a8cf0675e",
                "BacklogId": null,
                "WoNumber": "70156080",
                "UnitModel": "HD785-7",
                "SerialNumber": "9402",
                "UnitCode": "DT4112",
                "EquipmentNumber": "106921",
                "FormNumber": "HD785-7/3402/00/QA10/13(5)",
                "ZoneDesc": "ZONE 1 FRONT & UPPER AREA",
                "AreaDesc": "A. FRONT SIDE OF MACHINE CHECK",
                "ItemDesc": "Emergency Shutdown check condition",
                "Version": "1.00",
                "Measurement": "0",
                "Sequence": 7,
                "MPriority": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MNote": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MCondition": {
                    "Measurement": null,
                    "TypeDesc": "0",
                    "Code": "NMG",
                    "Desc": "Good",
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "Comment": null,
                "LeaderApprovedBy": "80101070@unitedtractors.com",
                "LeaderApprovedDate": "2020-04-21T10:02:03.293",
                "SupervisorApprovedDate": null,
                "SupervisorApprovedBy": null,
                "Id": "83eaa8ed-7ceb-53cf-ba9b-5c063a79bcfb",
                "CreatedBy": "81300006@unitedtractors.com",
                "CreatedDate": "2020-04-21T09:49:49.95",
                "UpdatedBy": "81300006@unitedtractors.com",
                "UpdatedDate": "2020-04-21T09:49:49.95",
                "IsActive": true
            },
            {
                "WoId": "47284214-faf8-464f-a967-d26a8cf0675e",
                "BacklogId": null,
                "WoNumber": "70156080",
                "UnitModel": "HD785-7",
                "SerialNumber": "9402",
                "UnitCode": "DT4112",
                "EquipmentNumber": "106921",
                "FormNumber": "HD785-7/3402/00/QA10/13(5)",
                "ZoneDesc": "ZONE 1 FRONT & UPPER AREA",
                "AreaDesc": "A. FRONT SIDE OF MACHINE CHECK",
                "ItemDesc": "LH Step Ladder check condition",
                "Version": "1.00",
                "Measurement": "0",
                "Sequence": 8,
                "MPriority": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MNote": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MCondition": {
                    "Measurement": null,
                    "TypeDesc": "0",
                    "Code": "NMG",
                    "Desc": "Good",
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "Comment": null,
                "LeaderApprovedBy": "80101070@unitedtractors.com",
                "LeaderApprovedDate": "2020-04-21T10:02:03.293",
                "SupervisorApprovedDate": null,
                "SupervisorApprovedBy": null,
                "Id": "7a631e78-9d05-5432-9253-2320b13c4ed9",
                "CreatedBy": "81300006@unitedtractors.com",
                "CreatedDate": "2020-04-21T09:49:49.959",
                "UpdatedBy": "81300006@unitedtractors.com",
                "UpdatedDate": "2020-04-21T09:49:49.959",
                "IsActive": true
            },
            {
                "WoId": "47284214-faf8-464f-a967-d26a8cf0675e",
                "BacklogId": null,
                "WoNumber": "70156080",
                "UnitModel": "HD785-7",
                "SerialNumber": "9402",
                "UnitCode": "DT4112",
                "EquipmentNumber": "106921",
                "FormNumber": "HD785-7/3402/00/QA10/13(5)",
                "ZoneDesc": "ZONE 1 FRONT & UPPER AREA",
                "AreaDesc": "A. FRONT SIDE OF MACHINE CHECK",
                "ItemDesc": "RH Step Ladder check condition",
                "Version": "1.00",
                "Measurement": "0",
                "Sequence": 9,
                "MPriority": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MNote": {
                    "Code": null,
                    "Desc": null,
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "MCondition": {
                    "Measurement": null,
                    "TypeDesc": "0",
                    "Code": "NMG",
                    "Desc": "Good",
                    "Sequence": null,
                    "Id": null,
                    "CreatedBy": null,
                    "CreatedDate": null,
                    "UpdatedBy": null,
                    "UpdatedDate": null,
                    "IsActive": false
                },
                "Comment": null,
                "LeaderApprovedBy": "80101070@unitedtractors.com",
                "LeaderApprovedDate": "2020-04-21T10:02:03.293",
                "SupervisorApprovedDate": null,
                "SupervisorApprovedBy": null,
                "Id": "50b21f93-a2ca-5f79-afaf-d7c3965b561d",
                "CreatedBy": "81300006@unitedtractors.com",
                "CreatedDate": "2020-04-21T09:49:49.968",
                "UpdatedBy": "81300006@unitedtractors.com",
                "UpdatedDate": "2020-04-21T09:49:49.968",
                "IsActive": true
            }
        ]
    }
]