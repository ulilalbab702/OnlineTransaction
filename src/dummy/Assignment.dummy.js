export const listOfMechanicDummy = [
    {
      "Username": "string",
      "FirstName": "string",
      "LastName": "string",
      "Email": "string",
      "MechanicUT": true,
      "Jobs": [
        {
          "JobType": "string",
          "Total": 0
        }
      ]
    }
  ]
export const jobSummaryDummy = {
  "Date": "2020-06-04T16:18:05.5472055",
  "CreatedBy": "AGUS LISANTO",
  "AssignedBy": "AGUS LISANTO",
  "MechanicLeader": {
      "UserName": "82107038@unitedtractors.com",
      "Name": "SUPRIYADI SUPRIYADI"
  },
  "Mechanics": [],
  "HandoverDate": null,
  "HandoverCreatedBy": null,
  "HandoverAssignedBy": null,
  "HandoverMechanicLeader": null,
  "HandoverMechanics": null,
  "Job": {
      "WoId": "8ff577d1-2911-4bbd-9d55-ffad4fe76fc5",
      "WoNumber": "71018650",
      "JobType": "INS",
      "UnitModel": "HD465-7",
      "SerialNumber": "15163",
      "UnitCode": "HT3022",
      "EquipmentNumber": "71506",
      "IsHandover": false,
      "Id": "a162bd06-6aca-499a-9f97-a5839b8a0612",
      "CreatedBy": "80107002@unitedtractors.com",
      "CreatedDate": "2020-06-04T16:18:05.5472055",
      "UpdatedBy": "80107002@unitedtractors.com",
      "UpdatedDate": "2020-06-04T16:18:05.5472055",
      "IsActive": true
  }
}