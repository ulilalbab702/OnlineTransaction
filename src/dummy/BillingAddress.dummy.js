export const dataBillingAddressDummy = [
    {
        "fullName": "Test Integrasi",
        "contactNumber": "088122345",
        "email": "testintegrasi@unitedtractors.com",
        "position": "string",
        "addressLabel": "Kantor PT Global Service Indonesia",
        "city": "Jakarta",
        "address": "Jl Raya kaliabang Tengah, Ruko San Jaya, Blok A5 No 9,\r\nSebalah Indomaret",
        "postalCode": "13920",
        "locationLatitude": "3.669276961596295",
        "locationLongitude": "102.4003031491451",
        "isPrimary": true
    }]