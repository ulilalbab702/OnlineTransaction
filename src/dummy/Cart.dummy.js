export const cartDummyList = {
    "cartProductResponse": {
        "data": [
            {
                "cartId": "d7e23521-0d23-4f37-aceb-77f8f75b3d16",
                "productId": "32bdd633-932f-465a-b93c-219a70971008",
                "name": "600-311-8291",
                "imageThumbnailUrl": "https://untr-my.sharepoint.com/:i:/g/personal/cst_dev04_unitedtractors_com/EW__sgepd2xJs1m7uxNquHQBbRKmqLZ3DkmR7N1Mb7JWww?e=1f9bRa",
                "qty": 3,
                "priceAfter": 300000,
                "priceBefore": 500000,
                "currency": "IDR",
                "cookiesData": ""
            },
            {
                "cartId": "10d112f9-0a0b-4c26-87b6-87ffb0374a77",
                "productId": "4211034d-8571-4ed6-84b2-8b380443bf7b",
                "name": "568002",
                "imageThumbnailUrl": "https://untr-my.sharepoint.com/:i:/g/personal/cst_dev04_unitedtractors_com/EW__sgepd2xJs1m7uxNquHQBbRKmqLZ3DkmR7N1Mb7JWww?e=1f9bRa",
                "qty": 2,
                "priceAfter": 800000,
                "priceBefore": 1000000,
                "currency": "IDR",
                "cookiesData": ""
            }
        ],
        "meta": {
            "totalItems": 2,
            "pageNumber": 1,
            "pageSize": 5,
            "totalPages": 1,
            "hasPreviousPage": false,
            "hasNextPage": false,
            "nextPageNumber": 1,
            "previousPageNumber": 1
        }
    },
    "subTotalPrice": 2500000
}
