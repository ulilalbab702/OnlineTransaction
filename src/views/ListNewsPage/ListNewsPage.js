import React, { useEffect, useRef, useState } from 'react';
import {
  Row,
  Col,
  Button,
} from 'reactstrap';
import './style/index.scss';
import {
  artikel,
  imageNotFound
} from "assets/images";
import Breadcrumb from 'components/Breadcrumb/Breadcrumb';
import firebase from '../../firebase/firebase';
import parse from 'html-react-parser';
import { MENU } from 'constants/menu';

const ListNewsPage = (props) => {
  const [value, setValue] = useState("");
  const [page, setPage] = useState(1);

  const setAnalyticClevertap = (action, event, screen, product) => {
    if (props.user) {
      const { userName } = props.user;
      if (product !== null) {
        window.clevertapEventProduct(`${product[0] + "_LogedIn"}`, product[1]);
      } else {
        window.clevertapEvent(`${event + "_LogedIn"}`, props.user);
      }
      firebase.analytics().setUserId(userName);
      firebase.analytics().setUserProperties(userName, "username:" + userName + `||View: ${screen}`);
      firebase.analytics().setCurrentScreen(screen, screen);
      firebase.analytics().logEvent(`${event + "_LogedIn"}`);
      firebase.analytics().setUserProperties("username", userName);
      firebase.analytics().setUserProperties(action, `${event + "_LogedIn"}`);
    } else {
      if (product !== null) {
        window.clevertapEventProduct(`${product[0] + "_NotLogedIn"}`, product[1]);
      } else {
        window.clevertapEvent(`${event + "_NotLogedIn"}`, null);
      }
      firebase.analytics().setCurrentScreen(screen, screen);
      firebase.analytics().logEvent(`${event + "_NotLogedIn"}`);
      firebase.analytics().setUserProperties(action, `${event + "_NotLogedIn"}`);
    }
  };

  useEffect(() => {
    setAnalyticClevertap("view", "View_ListNews_Screen", "View_ListNews_Screen", null);
    window.scrollTo(0, 0);
  }, []);

  useEffect(() => {
    async function fetchData() {
      const response = await props.fetchListNews('', 6, page)
    }
    fetchData();
  }, []);

  useEffect(() => {
    async function fetchReq() {
      props.fetchListNewsReq()
    }
    fetchReq()
  }, [])

  const _renderShowMore = () => {
    if (props.dataNewsAll.hasNextPage) {
      return (
        <div onClick={() => {
          setPage(page + 1);
          setAnalyticClevertap("click", "Click_SeeMoreNews", "View_ListNews_Screen", null);
          props.fetchListNews('', 6, page + 1);
        }}
          className='cardMore' style={{ height: '50px', cursor: 'pointer' }}>
          <p className='titleButtonMore'>Lihat Berita Lainnya</p>
        </div>
      )
    }
  }

  const _renderBreadcrumb = () => {
    const breadcrums = [
      {
        'url': '/Landing',
        'name': 'Landing'
      },
    ];
    return (
      <div style={{ marginBottom: '1rem', marginRight: 'auto' }}>
        <Breadcrumb
          linkBreadcrumb={breadcrums}
          typography={"List berita"}
        />
      </div>
    )
  };

  const _renderNotFound = () => {
    return (
      <div className='containerNewsNotFound'>
        <img src={imageNotFound} className='imgNewsNotFound'></img>
        <p className='titleNewsNotFound'>Tidak bisa menemukan berita</p>
      </div>
    )
  };

  return (
    <div className="paddingContentListNews" style={{ paddingTop: 110 }}>
      {_renderBreadcrumb()}
      {props.listNewsData.data.length > 0 ?
        (<div className="gridNews">
          {props.listNewsData.data.map((item) => {
            return (
              <div
                key={item.newsId}
                className='cardListNews'
                onClick={() => {
                  setAnalyticClevertap("click", "Click_NewsDetail", "View_ListNews_Screen", ["Click_NewsDetail", { "News_Name": item.title }]);
                  props.push(`${MENU.DETAIL_PROMOTION_LIST}/${item.newsId}`)
                }}
              >
                <img className='imageNews' src={item.imageNewsUrl} />
                <Row className='contentNews' style={{ flexGrow: '0' }}>
                  <div className="contentDescription">
                    <p className='titleNews'>{item.title}</p>
                    <p className='newsDescription'>{parse(!item.description ? '-' : item.description.length > 100 ? `${item.description.substring(0, 100)}...` : item.description)}</p>
                  </div>
                </Row>
              </div>
            )
          })
          }
        </div>)
        :
        _renderNotFound()
      }
      <div style={{ display: 'flex', justifyContent: 'center' }}>
        {_renderShowMore()}
      </div>
    </div>
  )
}

export default ListNewsPage;