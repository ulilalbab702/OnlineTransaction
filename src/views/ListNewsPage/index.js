import ListNewsPage from './ListNewsPage';
import { connect } from 'react-redux';
import { LIST_NEWS_REQ } from 'actions/actionTypes';
import { listNews, listNewsReq } from 'actions/listNews'
import { push } from 'connected-react-router';
import {
    createLoadingSelector,
} from 'utils/selector.helper'

//Set selector loading 
const loadingSelector = createLoadingSelector([LIST_NEWS_REQ]);
const mapStateToProps = state => ({
    ...state.user,
    listNewsData: state.listNews.listNews.data,
    dataNewsAll: state.listNews.listNews,
    isLoading: loadingSelector(state)
});

const mapDispatchToProps = dispatch => ({
    push: (url, item) => dispatch(push(url, item)),
    fetchListNews: (attributValue, PageSize, PageNumber) => dispatch(listNews(attributValue, PageSize, PageNumber)),
    fetchListNewsReq: () => dispatch(listNewsReq())
});

export default connect(mapStateToProps, mapDispatchToProps)(ListNewsPage);