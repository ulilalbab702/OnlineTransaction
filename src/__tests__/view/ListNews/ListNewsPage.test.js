import React from "react";
import toJson from "enzyme-to-json";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import ListNewsPage from "views/ListNewsPage/ListNewsPage";
import { listNewsDummy, detailNews } from "./../../../dummy/ListNews.dummy";

Enzyme.configure({ adapter: new Adapter() });
describe("List News Component", () => {
    const wrapper = shallow(
        <ListNewsPage
            listNewsData={listNewsDummy}
            dataNewsAll={detailNews}
        />
    );
    describe("List News Component", () => {
        it("Should match to snapshot", () => {
            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });
});