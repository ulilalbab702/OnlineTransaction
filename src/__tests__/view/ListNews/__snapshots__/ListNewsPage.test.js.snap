// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List News Component List News Component Should match to snapshot 1`] = `
<div
  className="paddingContentListNews"
>
  <div
    className="gridNews"
  >
    <div
      className="cardListNews"
      key="daaa19b6-d0a6-460c-12c1-08d8d4877ae9"
      onClick={[Function]}
    >
      <img
        className="imageNews"
        src="https://firebasestorage.googleapis.com/v0/b/customer-portal-dan-ut-portal.appspot.com/o/Images%2F307949f7-1787-4ff9-97a4-410c6687d037%2F.jpg?alt=media&token=368a6249-238f-4c67-8352-a800e29c415a"
      />
      <Row
        className="contentNews"
        style={
          Object {
            "flexGrow": "0",
          }
        }
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="contentDescription"
        >
          <p
            className="titleNews"
          >
            Aplikasi Perawatan Alat Berat UT Connect, Mudahkan Akses Informasi Produk United Tractors
          </p>
          <p
            className="newsDescription"
          >
            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the ...
          </p>
        </div>
      </Row>
    </div>
    <div
      className="cardListNews"
      key="cb678399-9d98-4b60-12c4-08d8d4877ae9"
      onClick={[Function]}
    >
      <img
        className="imageNews"
        src="https://firebasestorage.googleapis.com/v0/b/customer-portal-dan-ut-portal.appspot.com/o/Images%2F307949f7-1787-4ff9-97a4-410c6687d037%2F.jpg?alt=media&token=368a6249-238f-4c67-8352-a800e29c415a"
      />
      <Row
        className="contentNews"
        style={
          Object {
            "flexGrow": "0",
          }
        }
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="contentDescription"
        >
          <p
            className="titleNews"
          >
            Aplikasi Perawatan Alat Berat UT Connect, Mudahkan Akses Informasi Produk United Tractors
          </p>
          <p
            className="newsDescription"
          >
            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the ...
          </p>
        </div>
      </Row>
    </div>
    <div
      className="cardListNews"
      key="9871434b-46ff-417d-12c2-08d8d4877ae9"
      onClick={[Function]}
    >
      <img
        className="imageNews"
        src="https://firebasestorage.googleapis.com/v0/b/customer-portal-dan-ut-portal.appspot.com/o/Images%2F307949f7-1787-4ff9-97a4-410c6687d037%2F.jpg?alt=media&token=368a6249-238f-4c67-8352-a800e29c415a"
      />
      <Row
        className="contentNews"
        style={
          Object {
            "flexGrow": "0",
          }
        }
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <div
          className="contentDescription"
        >
          <p
            className="titleNews"
          >
            Aplikasi UT Connect Telah hadir di IOS
          </p>
          <p
            className="newsDescription"
          >
            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the ...
          </p>
        </div>
      </Row>
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  />
</div>
`;
